### PR TITLE
 azurerm_storage_account - refactoring to use parsers

### PR DESCRIPTION
**Pull request by catriona-m**

### CODE REVIEW

File: internal/services/storage/storage_management_policy_resource.go [hunk 1]
```diff
@@ -222,22 +225,22 @@ func resourceStorageManagementPolicy() *pluginsdk.Resource {
 
 func resourceStorageManagementPolicyCreateOrUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).Storage.ManagementPoliciesClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
 	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	storageAccountId := d.Get("storage_account_id").(string)
```
Comment (stephybun) [inline internal/services/storage/storage_management_policy_resource.go:232]: I think we could omit this intermediate variable.

File: internal/services/storage/storage_management_policy_resource.go [hunk 2]
```diff
@@ -251,17 +254,11 @@ func resourceStorageManagementPolicyCreateOrUpdate(d *pluginsdk.ResourceData, me
 		},
 	}
 
-	result, err := client.CreateOrUpdate(ctx, resourceGroupName, storageAccountName, parameters)
-	if err != nil {
+	if _, err := client.CreateOrUpdate(ctx, rid.ResourceGroup, rid.Name, parameters); err != nil {
 		return fmt.Errorf("creating Azure Storage Management Policy %q: %+v", storageAccountId, err)
```
Comment (stephybun) [inline internal/services/storage/storage_management_policy_resource.go:258]:
```suggestion
		return fmt.Errorf("creating %s: %+v", mgmtPolicyId, err)
```

**Review by stephybun — APPROVED** [review]
Thanks @catriona-m, left a few suggestions other LGTM 👍🏼

File: internal/services/storage/storage_management_policy_resource.go (hunk 1 above)
Comment (stephybun) [inline internal/services/storage/storage_management_policy_resource.go:232]: I think we could omit this intermediate variable.

Comment (stephybun) [reply internal/services/storage/storage_management_policy_resource.go:232]: GH is messing with me, this comment is supposed to refer to [line 232](https://github.com/hashicorp/terraform-provider-azurerm/blob/8ac01c6e9d5de3dc04c3642e984f5d435f095133/internal/services/storage/storage_management_policy_resource.go#L232)

Comment (catriona-m) [reply internal/services/storage/storage_management_policy_resource.go:232]: turns out subscriptionId can also go too anyway!

**Pull request merged**